### PR TITLE
bpo-41690: Use a loop to collect args in the parser instead of recursion

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue41690

### CODE REVIEW

File: Parser/pegen.c [hunk 1]
```diff
@@ -2217,3 +2217,38 @@ _PyPegen_nonparen_genexp_in_call(Parser *p, expr_ty args)
...
+    }
+
+    asdl_seq *starreds = _PyPegen_seq_extract_starred_exprs(p, b);
+    asdl_seq *keywords = _PyPegen_seq_delete_starred_exprs(p, b);
+
+    if (starreds) {
+        total_len += asdl_seq_LEN(starreds);
+    }
+
+    asdl_seq *new_seq = _Py_asdl_seq_new(total_len, p->arena);
```
Comment (gvanrossum) [inline Parser/pegen.c]: I'd rename this `new_args` or even `args`.

File: Parser/pegen.c [hunk 2]
```diff
@@ -2217,3 +2217,38 @@ _PyPegen_nonparen_genexp_in_call(Parser *p, expr_ty args)
...
+
+    Py_ssize_t i = 0;
+    for (i = 0; i < args_len; i++) {
+        asdl_seq_SET(new_seq, i, asdl_seq_GET(a, i));
+    }
+    for (;i < total_len; i++) {
```
Comment (gvanrossum) [inline Parser/pegen.c]: Surely PEP7 wants a space after the first `;`.

**Review by gvanrossum — COMMENTED** [review]
Just style nits -- should I file a bpo bug to track this?

It's awkward that this is happening so close to rc2. But it's technically a regression from 3.8, and a legit way to cause a segfault without even running code. Plus for N=20000 you can really feel the quadratic time.

@ambv We'd like this to be adopted into rc2 (in about 10 days).

**Comment by pablogsal** [comment]
Opened https://bugs.python.org/issue41690 to track this

**Review by gvanrossum — APPROVED** [review]
Go!

**Pull request merged**